e2e/cypress/integration/messaging/date_separator_spec.js

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+import users from '../../fixtures/users.json';
+
+const sysadmin = users.sysadmin;
+
+describe('Messaging', () => {
+    let newChannel;
+
+    before(() => {
+        // # Login as user-1 and set preferences for locale and timezone
+        cy.apiLogin('user-1');
+        cy.apiPatchMe({
+            locale: 'en',
+            timezone: {automaticTimezone: '', manualTimezone: 'UTC', useAutomaticTimezone: 'false'},
+        });
+
+        // # Create and visit new channel
+        cy.createAndVisitNewChannel().then((channel) => {
+            newChannel = channel;
+        });
+    });
+
+    it('MM-21482 Date separators should translate correctly', () => {
+        function verifyDateSeparator(index, match) {
+            cy.findAllByTestId('basicSeparator').eq(index).within(() => {
+                cy.findByText(match);
+            });
+        }
+
+        // # Post a message with old date
+        const oldDate = Date.UTC(2020, 0, 5, 12, 30); // Jan 5, 2020 12:30pm
+        cy.postMessageAs({sender: sysadmin, message: 'Hello from Jan 5, 2020 12:30pm', channelId: newChannel.id, createAt: oldDate});
+
+        // # Post message from yesterday
+        const yesterdaysDate = Cypress.moment().subtract(1, 'days').valueOf();
+        cy.postMessageAs({sender: sysadmin, message: 'Hello from yesterday', channelId: newChannel.id, createAt: yesterdaysDate});
+
+        // # Post a message for today
+        cy.postMessage('Hello from today');
+
+        // # Reload to re-arrange post order
+        cy.reload();
+
+        // * Verify that the date separators are rendered in English
+        verifyDateSeparator(0, /^(Sat|Sun), Jan (04|05), 2020/);
```

Comment:
The accuracy of date is not being tested here since it's done in the library itself.  We're much concern on short date format. The variation is dependent on server and Cypress timezone in prod, test and local environment.